### ISSUE TITLE
Improve accessibility of help text

### ISSUE THREAD

**Issue by justcolin** [projecthydra/sufia#2003]
### Descriptive summary

In our user and accessibility testing we found that the default presentation of help text on Sufia forms is making some things difficult for users. Issues include (in loose order of importance):

- The help buttons to show/hide the help text do not have visible text and they are very small, making it very difficult for users using voice control software to select them.
    - Here is an example of the annoying process someone using voice control software has to go through to move the mouse: https://youtu.be/G4zUOwP3F_k?t=57s
    - The rule of thumb is to avoid making users go through that process whenever possible.
    - When buttons have visible text users of voice control software can often just tell the computer to click the button by name, but in this case I think that would clutter the screen with repeated link/button names, which is an accessibility problem for other users.
- The help buttons are so small they are a bit tricky for most users to select with a mouse, and would be very difficult for users with motor disabilities to select.
    - Some users with motor disabilities use touch screens, another reason why even desktops applications should prefer larger buttons.
- Many users have been confused about how to close the help boxes.
- It is not obvious to users who are less computer literate that there is help text available.
    - As a university with many older faculty, we can't assume that every user is going to be great using fancier interfaces.
- The modals/pop-ups cover up parts of the interface that we are trying to help people with.
    - This is especially problematic for people who are not as computer literate.

### Suggested fix

Put the help text bellow the label, visible to everyone. It solves all the accessibility problems, is a lot simpler to maintain, and is standard across most websites so users shouldn't find the added text jarring.

When people need to add a lot of help text, we recommend that they write a short 1–2 sentence summary of what the first does and include a link to a static help page in that description for a longer explanation about that field.

Here is an example of how things can look (screenshot taken from U-M's research data repository, so some things are different from the Sufia defaults):

![screen shot 2016-05-09 at 3 59 35 pm copy](https://cloud.githubusercontent.com/assets/11234702/15126562/89a2f09e-1600-11e6-88c3-056e0323c6f4.png)

### Questions

1. Are people interested in a "Hide Help Text" button at the top of each form or as an option in a user's profile page?

2. Is anyone in the community interested in a configuration option to bring back the less accessible help text buttons for institutions who prefer it? If so I am more than happy to add that feature to the PR.

3. If people are interested in the the option mentioned in question 2, would they be okay with some accessibility improvements to the help button interface? This will probably consist of a little visible text saying to make it easier for people using voice control software, as well as a dedicated button to close the popup.

### Related work

#1612

**Comment by justcolin** [comment]
I am currently working on a PR for this.

Does anyone in @projecthydra/sufia-ui-ux-advisors have any thoughts?

**Comment by heidisb** [comment]
Colin's summary provides a lot of evidence that hiding help text under help buttons/icons is not the best. Beyond the accessibility issues and concrete problems we've found...I also encourage going back to what many consider the #1 usability principle: don't make users think. 

By having the help text hidden under the button you:
1. Make them think (extra hard) about what they need to enter in each field
2. Make them think about what is under the question mark button/icon or if it's clickable at all (if they even notice it)
3. Make them think about how to close the pop-up

I do not think putting help text under a button/icon like this is inherently bad, but in the context of the upload screen where needing some level of direction at some point is more or less inevitable, it really does not make sense to me here. Typically, my thought with ? buttons/icons is that they are more about "what is this?" than "what do I do?"...if that helps.

As someone who led writing the help text for our RDR upload page, if said text is written for the web and done well, you can provide essential information without overloading the page. And as Colin mentions, you can also layer the info and provide a link to a more detailed help page as necessary.

Heidi

**Comment by sfbetz** [comment]
+1 to this. We had *exactly* the same results as you in user testing we did last summer. Our solution was the same as you suggest: we reduced the amount of "help" text, and embedded it right in the deposit form instead of hiding it behind buttons. People were much less confused about how to fill in the deposit form after this change.  My thoughts on your questions:

1. I don't think "hide help text" is necessary. Once someone is familiar with an interface, they'll just ignore the help text and populate the fields. 
2. Pretty sure that consistent test results of users having issues with help buttons would be a good argument for not bringing them back, but maybe there are people out there who have customized buttons in some way?

**Comment by mtribone** [comment]
@justcolin I really like the context of +Add and the label on the button (+Add Contributor). I've always seen the help icon all over the page as a problem, so adding minimal help text is a good solution. I worry that others will get too wordy, but I guess that is the responsibility of the adopter if they want to override the text.

I'm curious on how to approach the tome that is the rights metadata. If we link to another page does the user lose anything by moving back and forth? By using a modal, we keep the context of  filling out the form. Similar feeling about the keywords in the facets and using a modal, so looking forward to seeing your approach.

I agree with @sfbetz that hiding help text is unnecessary as a configuration option.

**Comment by newmanld** [comment]
I agree with the gist of comments here.  Embedding very brief help text in context in the form is a solution we've used locally.  A button to 'hide help text' is unnecessary.

**Comment by justcolin** [comment]
@sfbetz, @newmanld and @mtribone: Thanks for the feedback!

As for the rights metadata: I don't like the modal because it still covers up the form, and unlike pulling it open in a new window or tab, the modal it doesn't save where you were scrolled to if you switch between it and the form. I agree with @newmanld that it should be it's own issue. Maybe the PR I am working on for this issue can just that info into a separate static page, then we can discuss further enhancements in another issue?

In our current interface our quick solution was to put a short description next to each option, then we provided a link to the CC documentation. The styling/padding could use some improvement, but it provided an easy quick fix:

![screen shot 2016-05-10 at 11 51 50 am](https://cloud.githubusercontent.com/assets/11234702/15153050/b86222ce-16a5-11e6-884e-3d1efc6883eb.png)

**Comment by justcolin** [comment]
@mtribine: I am glad to hear you like the added context on the add buttons, you may like an issue/PR I'll be posting later. ;-) We've done some user testing and have come up with a number of improvements to the "Add" buttons which seem to help users, so stay tuned.

**Comment by mtribone** [comment]
Sufia by default has the following and some people have asked about GPL. We're gonna need a bigger boat. @newmanld Very interested in your solution!

![screen shot 2016-05-10 at 12 46 06 pm](https://cloud.githubusercontent.com/assets/4163828/15154762/43f09076-16ad-11e6-9f02-2162712b567c.png)

**Comment by sfbetz** [comment]
Would LOVE a license selector tool - choosing a license is a serious barrier for researchers completing self-deposits. CC makes it easy: https://creativecommons.org/choose/ - But our needs are more complex than that (eg. need options for adding publisher statements, etc). Would be happy to work with someone on this in future!

**Comment by crowesn** [comment]
We're working on implementing the [Public License Selector](https://github.com/ufal/public-license-selector) at UC, for our curate based app at UC. We're customizing the questions to be more work oriented.

**Comment by justcolin** [comment]
@projecthydra/sufia-ui-ux-advisors: I am creating a new issue to talk about the license selection issue since that is a separate topic, and I have some accessibility/usability concerns there.

Any other comments on the help text before I finish up a PR?

**Comment by justcolin** [comment]
See issue #2007

**Comment by justcolin** [comment]
I'll see about getting a PR ready by the end of the week.

**Comment by newmanld** [comment]
@mtribone - Re our local solution - the developer currently working on this agrees that modal is problematic for mobile and touch devices. We’ve tentatively decided to hide the chooser (rights dialogue) below a preset screen resolution.   I'll share screen shots when we have something to show (or comment if we back away from this approach).

**Comment by mjgiarlo** [comment]
@justcolin When do you think a PR might be available for this work? I'm considering including this in 7.0.0.

**Comment by justcolin** [comment]
@mjgiarlo: I should have it done next week. Would that work for you?

The work is mostly done, I am just making sure things work well in Sufia and getting @heidisb to help out with some potential improvements to the existing help text (along with also spending some time on other projects, thus the slower development time on this).

**Comment by justcolin** [comment]
We may also have the add/remove button work done by then as well.

**Comment by mjgiarlo** [comment]
Works for me, @justcolin.

**Comment by mjgiarlo** [comment]
@justcolin have any work you'd like folks to review this week?

**Comment by justcolin** [comment]
I am trying to get that out for you today. In the middle of a refactor currently.

**Comment by mjgiarlo** [comment]
Much obliged!

**Comment by justcolin** [comment]
@mjgiarlo: I have a PR ready for CC **except** the visibility field needs some improvements. I have some suggested changes from @heidisb, but it will take a little longer to fix up the myriad of things that render that field and package it all up into a PR. Would you like the CC help text changes now or to package all of the changes together into one PR?

**Comment by justcolin** [comment]
@mjgiarlo: I also almost have the set of changes Sufia will need to work with that CC PR ready, but the rights field help text requires some touching up first to prevent errors.

**Comment by mjgiarlo** [comment]
@justcolin Either way works for me. Thanks!

**Comment by justcolin** [comment]
This issue can be closed as soon as #2085 is fixed.

**Comment by justcolin** [comment]
Fixed by #2103
